### PR TITLE
Configure Eslint Warnings to fail CI

### PR DESCRIPTION
**Pull request by topocount**
This change updates the repo lint:check script to return an error state
when one or more warnings are encountered.

Nothing else changes, but this error state will cause CI to fail when
warnings are encountered, ideally forcing devs to address the lint
problems so they don't get merged to main

### CODE REVIEW

**Comment by topocount** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1707** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1707" target="_blank">Graphite</a> 
    * **PR #1708** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1708" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1708?utm_source=stack-comment).

**Review by levity — APPROVED** [review]

**Comment by topocount** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1708?utm_source=stack-comment).


Job Reference: yiT5HuLaAWTCkDS5BiSD

**Pull request merged**
